### PR TITLE
Ftr/Transfer tracing context for dubbo protocol

### PR DESCRIPTION
**Pull request by flycash**
**What this PR does**:
We can transfer the tracing context end to end now

### CODE REVIEW

File: common/constant/key.go [hunk 1]
```diff
@@ -78,6 +78,7 @@ const (
 
 const (
 	DUBBOGO_CTX_KEY = "dubbogo-ctx"
+	CONTEXT_KEY     = "context"
```
Comment (zouyx) [inline common/constant/key.go]: what is that key use for ? it should add some comment

Comment (flycash) [reply common/constant/key.go]: It's not used, so I remove it.

File: protocol/dubbo/listener.go [hunk 2]
```diff
@@ -327,6 +331,22 @@ func (h *RpcServerHandler) OnCron(session getty.Session) {
 	}
 }
 
+// rebuild the context by attachment.
```
Comment (AlexStocks) [inline protocol/dubbo/listener.go]: as go lint said, the func name should be the first word of its comment.

File: protocol/dubbo/listener.go [hunk 3]
```diff
@@ -327,6 +331,22 @@ func (h *RpcServerHandler) OnCron(session getty.Session) {
...
+// Once we decided to transfer more context's key-value, we should change this.
+// now we only support rebuild the tracing context
+func rebuildCtx(inv *invocation.RPCInvocation) context.Context {
+	ctx := context.Background()
+
+	// actually, if user do not use any opentracing framework, it will always be error.
```
Comment (AlexStocks) [inline protocol/dubbo/listener.go]: it will always be error. --> the error will not be nil.

File: protocol/dubbo/listener.go [hunk 4]
```diff
@@ -327,6 +331,22 @@ func (h *RpcServerHandler) OnCron(session getty.Session) {
...
+	spanCtx, err := opentracing.GlobalTracer().Extract(opentracing.TextMap,
+		opentracing.TextMapCarrier(inv.Attachments()))
+
```
Comment (AlexStocks) [inline protocol/dubbo/listener.go]: delete this blank line.

**Comment by AlexStocks** [comment]
LGTM

**Pull request merged**